src/qt/walletcontroller.cpp

```diff
@@ -31,7 +31,7 @@ WalletController::WalletController(interfaces::Node& node, const PlatformStyle*
     , m_options_model(options_model)
 {
     m_handler_load_wallet = m_node.handleLoadWallet([this](std::unique_ptr<interfaces::Wallet> wallet) {
-        getOrCreateWallet(std::move(wallet));
+        if (wallet) getOrCreateWallet(std::move(wallet));
```

Comment:
In what case `wallet` is null?

Reply:
See line 95 in `qt/walletcontroller.cpp`
There it checks if the wallet is empty even though it isn't null.

Reply:
@promag In case you call `loadwallet` very early in `bitcoin-qt` init. @fanquake has written a python script in https://github.com/bitcoin/bitcoin/issues/16937 that triggers it consistently.

Reply:
This fix is incorrect.

Reply:
Okay. Care to elaborate?

Reply:
Sorry, forgot to update this, see #16963.

Note that conceptually you are being notified about a new wallet, it should exist! Also this change hides a more severe problem which is explained in my PR.

Reply:
Thanks, closing this now in favour of https://github.com/bitcoin/bitcoin/pull/16963.